src/error.rs

```diff
@@ -52,24 +73,41 @@ impl ErrorKind {
     /// A description of this error kind
     pub fn description(self) -> &'static str {
         match self {
-            ErrorKind::Unavailable => "permanent failure",
+            ErrorKind::Unavailable => "permanently unavailable",
+            ErrorKind::Failure => "failure",
             ErrorKind::Transient => "transient failure",
             ErrorKind::NotReady => "not ready yet",
-            ErrorKind::Other => "uncategorised",
+            ErrorKind::Custom(_) => "custom error code",
```

Comment:
This method can only return a static string and it's not guaranteed that we have a cause.

Reply:
That is an argument. Reminds me a lot of that experiment I did with keeping the OS errno we decided against, because we don't assume an OS with `no_std` (https://gist.github.com/pitdicker/2411a4e3aae245c217e4fdd8a0b99917).

Like you suggested and @vks said, I would wait until people start asking for this solution.